File: 2309-greatest-english-letter-in-upper-and-lower-case/2309-greatest-english-letter-in-upper-and-lower-case.py
class Solution:
    def greatestLetter(self, s: str) -> str:
        seen, res = set(), ""
        for c in s:
            seen.add(c)
            up = c.upper()
            low = c.lower()
            if up in seen and low in seen:
                if res == "" or res < up:
                    res = up
        return res

File: 2309-greatest-english-letter-in-upper-and-lower-case/2309-greatest-english-letter-in-upper-and-lower-case.js
/**
 * @param {string} s
 * @return {string}
 */
var greatestLetter = function(s) {
    const set = new Set()
    let res = ""
    for (const c of s) {
        set.add(c)
        const upper = c.toUpperCase()
        const lower = c.toLowerCase()
        if (set.has(upper) && set.has(lower)) {
            if (res === "" || res < upper) {
                res = upper
            } 
        }
    }
    return res
};